### PR TITLE
[FLINK-34970] Translate architecture documents into Chinese

### PR DESCRIPTION
**Pull request by caicancai**
## What is the purpose of the change
https://issues.apache.org/jira/browse/FLINK-34970
Translate architecture documents into Chinese

## Brief change log

*(for example:)*
  - *Periodic savepoint trigger is introduced to the custom resource*
  - *The operator checks on reconciliation whether the required time has passed*
  - *The JobManager's dispose savepoint API is used to clean up obsolete savepoints*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (yes / no)
  - Core observer or reconciler logic that is regularly executed: (yes / no)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by caicancai** [comment]
2024-03-31 11-33-01屏幕截图

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [inline docs/content.zh/docs/concepts/architecture.md:27]: It's not clear to me whether the title should be translated

Comment (RocMarshal) [reply docs/content.zh/docs/concepts/architecture.md:27]: > It's not clear to me whether the title should be translated

hi, @caicancai 
Thx for the pr. 

It would be better to translate it like other translated pages~

Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md]: Thank you for your reply, done

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:34]: ditto

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:50]: ditto

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:68]: ditto

**Review by RocMarshal — COMMENTED** [review]
Thanks @caicancai for  the update .
LGTM on the whole.
Just a few of minor comments.
PTAL  :)

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md:29]: a name="Architecture"></a> ？

Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md:29]: Thank you for your review, I have added it, please check if it is correct

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:27]:
```suggestion
<a name="architecture"></a>
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:36]:
```suggestion
<a name="control-loop"></a>
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:54]:
```suggestion
<a name="flink-resource-lifecycle"></a>
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (RocMarshal) [inline docs/content.zh/docs/concepts/architecture.md:74]:
```suggestion
<a name="admission-control"></a>
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:48]: Leave one space before&after English word,
```suggestion
2. Operator 观察 Flink 资源的当前状态（如果先前已部署）
3. Operator 验证提交的资源更改
4. Operator 协调任何必要的更改并执行升级
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:50]: 所需 -> 期望的

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:52]:
```suggestion
Operator 使用 [Java Operator SDK](https://github.com/java-operator-sdk/java-operator-sdk) 构建，并使用 [Native Kubernetes Integration](https://nightlies.apache.org /flink/flink-docs-master/docs/deployment/resource-providers/native_kubernetes/) 用于启动 Flink deployment 并在后台提交作业。
Java Operator SDK 是一个更高级别的框架和相关工具，用于支持使用 Java 编写 Kubernetes Operator。Java Operator SDK 和 Flink 的原生 kubernetes 集成本身都使用 [Fabric8 Kubernetes 客户端](https://github.com/fabric8io/kubernetes-client) 与 Kubernetes API 服务器交互。
```

`The Operator is built with the ` and `to interact with the Kubernetes API Server.` are missed here.

Also, how about leave deployment instead of `部署`?

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:58]:
```suggestion
Operator 管理 Flink 资源的生命周期。以下图表说明了可能的状态和转换：
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:66]:
```suggestion
  - SUSPENDED : （job）资源已被暂停
```

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:67]: 规范 -> 规则 or (Using Spec)

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:68]: `但尚未被视为稳定` -> `但仍不稳定`

翻译后的中文尽量通俗易懂，不用跟着 英文翻译。

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:71]: 规格

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:72]: 工作 -> job or 作业

**Review by 1996fanrui — COMMENTED** [review]
Thanks @caicancai for the great work!

Overall is good for me, i left some comments, please take a look in your free time, thanks a lot.

I wanna mention some common issues here due to we have a lot of pages will be translated in the future.

1. Can we keep some Terminology (do not translate them)? such as: operator, deployment, etc
2. We don't need to translate for every English word. Try to ensure that the translated Chinese is easy to understand and not awkward to read. (尽量保证翻译后的中文通俗易懂，读起来不别扭。)

**Comment by 1996fanrui** [comment]
Also, it's better to update the PR description. We can update these items or remove them if you think they are unneeded, thanks~

**Comment by caicancai** [comment]
> Thanks @caicancai for the great work!
> 
> Overall is good for me, i left some comments, please take a look in your free time, thanks a lot.
> 
> I wanna mention some common issues here due to we have a lot of pages will be translated in the future.
> 
> 1. Can we keep some Terminology (do not translate them)? such as: operator, deployment, etc
> 2. We don't need to translate for every English word. Try to ensure that the translated Chinese is easy to understand and not awkward to read. (尽量保证翻译后的中文通俗易懂，读起来不别扭。)

Thanks for the reminder, I will improve it

**Comment by caicancai** [comment]
cc @1996fanrui

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [reply docs/content.zh/docs/concepts/architecture.md]: > `当操作员接收到新的资源更新时，它将采取行动，以调整 Kubernetes 集群以达到所需状态，作为其协调循环的一部分。` -> `当 Operator 接收到新的资源更新时，它将调整 Kubernetes 集群以达到所需状态，这个调整将作为其协调循环的一部分。`

Is it not updated?

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [reply docs/content.zh/docs/concepts/architecture.md]: Sorry, it's a typo. It should be `规格`

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md]: Sorry, I forgot to submit

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [reply docs/content.zh/docs/concepts/architecture.md]: Is it not updated?

The `在使用 [Helm] 在 Kubernetes 集群上` seems a little wired.

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:57]: As https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications mentioned : `汉字与英文、数字之间需空格`.

The Specification should be respected for all translations.

Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md:57]: Thank you

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:31]: 1. `Flink 部署` -> `Flink deployments `
2. How about translate `schema` to `结构` instead of `模式`?
3. As https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications mentioned, `是一个集群范围的资源`, we don't need `一个` here.

image

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [inline docs/content.zh/docs/concepts/architecture.md:52]: Sorry, the second link has a space.

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [reply docs/content.zh/docs/concepts/architecture.md:57]: `Flink资源生命周期` still without space.

File: docs/content.zh/docs/concepts/architecture.md
Comment (1996fanrui) [reply docs/content.zh/docs/concepts/architecture.md]: Hi @caicancai , it still not updated?

Two '在' seems a little wired for me. Feel free to feedback here If you have any thoughts. 

My comments may not all be correct, and I'd like to hear your thoughts, thanks~

**Review by 1996fanrui — COMMENTED** [review]
Thanks @caicancai for the updated.

I believe this PR is close to being merged. I found some of old comments are not addressed, and I left some new comments as well.

Please take a look in your free time, thanks~

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md:31]: Sorry, I overlooked this, next time I should read it aloud to check

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md]: done

File: docs/content.zh/docs/concepts/architecture.md
Comment (caicancai) [reply docs/content.zh/docs/concepts/architecture.md]: Sorry, done

**Review by 1996fanrui — APPROVED** [review]
Thanks @caicancai for the hard work, LGTM assuming the CI is green.

Hi @RocMarshal , would you like to double check again before merging?

**Comment by caicancai** [comment]
squashed

**Pull request merged**
